tests/benchmark/benchmark.py

```diff
@@ -11,20 +11,36 @@
 
 
 def run_benchmark(args):
-    print("Generating dataset: {} rows * {} columns".format(args.rows, args.columns))
-    print("{}/{} test/train split".format(args.test_size, 1.0 - args.test_size))
-    tmp = time.time()
-    X, y = make_classification(args.rows, n_features=args.columns, random_state=7)
-    if args.sparsity < 1.0:
-       X = np.array([[np.nan if rng.uniform(0, 1) < args.sparsity else x for x in x_row] for x_row in X])
 
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=args.test_size, random_state=7)
-    print ("Generate Time: %s seconds" % (str(time.time() - tmp)))
-    tmp = time.time()
-    print ("DMatrix Start")
-    dtrain = xgb.DMatrix(X_train, y_train, nthread=-1)
-    dtest = xgb.DMatrix(X_test, y_test, nthread=-1)
-    print ("DMatrix Time: %s seconds" % (str(time.time() - tmp)))
+    try:
```

Comment:
Probably good idea to keep this version, which is much smarter about repeated operations.

Reply:
Better in xgboost than managing it in this separate branch.  If want to  keep original benchmark simple, can make separate benchmark_repeat.py that uses this code.